exercises/practice/tournament/.docs/instructions.append.md

```diff
@@ -0,0 +1,131 @@
+# placeholder
+
+## Sorting in AWK
+
+This exercise requires the output to be sorted in a particular way.  Like
+many programming languages, iterating over an associative array occurs in
+some random-looking internal hashing order of the keys. This is typically
+not what humans want to see.  GNU awk provides some sorting features.
+
+1. `asort` -- sort a numerically indexed array based on the array _values_.
+
+    ```awk
+    #!gawk 
+    BEGIN {
+        for (i = 1; i <= 5; i++)
+            nums[i] = int(100 * rand())
+
+        # returns the array size
+        n = asort(nums, sorted)
+
+        for (i = 1; i <= n; i++)
+            print i, nums[i], sorted[i]
+    }
+    ```
+    which outputs
+    ```none
+    1 92 30
+    2 59 57
+    3 30 59
+    4 57 74
+    5 74 92
+    ```
+
+    The form `n = asort(array)` sorts the named array in-place.
+
+    This is most useful for numerically-indexed arrays. The process that
+    populates the `sorted` array discards the originally arrays indices:
+    the `sorted` array is indexed from 1 to `n`.
+
+1. `asorti` -- sort any array based on the array _indices_.
+
+    The `asorti` function populates a "destination" array containing the 
+    "source" array's indices sorted.
+
+    ```awk
+    #!gawk
+    BEGIN {
+        nums["foo"] = int(100 * rand())
+        nums["bar"] = int(100 * rand())
+        nums["qux"] = int(100 * rand())
+
+        print "unordered"
+        for (i in nums)
+            print i, nums[i]
+
+        print "ordered by index"
+        n = asorti(nums, sorted)
+        for (i = 1; i <= n; i++)
+            print sorted[i], nums[sorted[i]]
+    }
+    ```
+    which outputs
+    ```none
+    unordered
+    foo 92
+    qux 30
+    bar 59
+    ordered by index
+    bar 59
+    foo 92
+    qux 30
+    ```
+
+1. `PROCINFO["sorted_in"]` -- sort any array using a sorting function.
+
+    The builtin PROCINFO array contains a lot of information about the
+    running awk process. One interesting element of this array is
+    `sorted_in`. The value of this array key is the name of a function
+    which controls the order in which array indices will be processed by `for
+    (indx in array)` loops.
+
+    The function can be the name of a user-defined function (_hint: this is
```

Comment:
Do you want to show the function signature it expects?

Reply:
Yes